docs/internals/ARCHITECTURE.md

```diff
@@ -27,8 +27,9 @@ performance under extreme contention.
 ## Overview
 
 TigerBeetle is a distributed system. It is a cluster of six replicas. The state of each replica is
-a single file on disk, called the data file. Replicas use a consensus protocol to ensure that their
-respective data files store identical data.
+a single file on disk, called the data file (see [./data_file.md](./data_file.md)). Replicas use a
```

Comment:
Is there a reason for doing this instead of that?
```suggestion
a single file on disk, called the [data file](./data_file.md). Replicas use a
```

Reply:
Yes! I want to show that:

* this leads to more internal docs rather than, say, a Wikipedia page (not super important for data file, but needed for consensus
* To hint the reader that there are more docs in the folder, and that they might want to read them.